src/core/server/opensearch/version_check/ensure_opensearch_version.ts

```diff
@@ -53,63 +52,64 @@ import type { OpenSearchClient } from '../client';
  * If the supplied node attribute is missing then we return null and use default fan out behavior
  * @param {OpenSearchClient} internalClient
  * @param {OptimizedHealthcheck} healthcheck
- * @returns {string|null} '_local' if all nodes have the same cluster_id, otherwise null
+ * @returns {string|string[]|null} '_local' if all nodes have the same cluster_id, array of node ids if different cluster_id, null if no cluster_id or nodes returned
  */
 export const getNodeId = async (
   internalClient: OpenSearchClient,
   healthcheck: OptimizedHealthcheck
-): Promise<string | null> => {
+): Promise<string | string[] | null> => {
```

Comment:
From the docstring - does it return a single string other than `_local`? If not,
`<'_local' | string[] | null>` might be better.